ranger-schema-registry-plugin/authorizer-shim/src/main/java/com/hortonworks/registries/ranger/authorization/schemaregistry/authorizer/RangerSchemaRegistryAuthorizer.java

```diff
@@ -0,0 +1,238 @@
+package com.hortonworks.registries.ranger.authorization.schemaregistry.authorizer;
+
+import org.apache.ranger.plugin.classloader.RangerPluginClassLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import java.util.Set;
+
+public class RangerSchemaRegistryAuthorizer implements Authorizer {
+
+    private static final Logger LOG  = LoggerFactory.getLogger(RangerSchemaRegistryAuthorizer.class);
+
+    private static final String   RANGER_PLUGIN_TYPE                      = "schema-registry";
+    private static final String   RANGER_SR_AUTHORIZER_IMPL_CLASSNAME  =
+            "com.hortonworks.registries.ranger.authorization.schemaregistry.authorizer.RangerSchemaRegistryAuthorizerImpl";
+
+    private Authorizer  rangerSRAuthorizerImpl;
+    private static RangerPluginClassLoader rangerPluginClassLoader;
+
+    public RangerSchemaRegistryAuthorizer() {
+        if(LOG.isDebugEnabled()) {
+            LOG.debug("==> RangerSchemaRegistryAuthorizer.RangerSchemaRegistryAuthorizer()");
+        }
+
+        this.init();
+
+        if(LOG.isDebugEnabled()) {
+            LOG.debug("<== RangerSchemaRegistryAuthorizer.RangerSchemaRegistryAuthorizer()");
+        }
+    }
+
+    private void init(){
+        if(LOG.isDebugEnabled()) {
+            LOG.debug("==> RangerSchemaRegistryAuthorizer.init()");
+        }
+
+        try {
+
+            rangerPluginClassLoader = RangerPluginClassLoader.getInstance(RANGER_PLUGIN_TYPE, this.getClass());
+
+            @SuppressWarnings("unchecked")
+            Class<Authorizer> cls = (Class<Authorizer>) Class.forName(RANGER_SR_AUTHORIZER_IMPL_CLASSNAME, true, rangerPluginClassLoader);
+
```

Comment:
Why are we using a separate classloader? I believe the ranger plugin is expected in the classpath anyway. You should use reflection and just retain the instance.
https://github.com/apache/ranger/blob/d6e0e9af2446d023e68df1e21221d8bf1190615d/plugin-kafka/src/main/java/org/apache/ranger/authorization/kafka/authorizer/RangerKafkaAuthorizer.java

Reply:
It is needed because ranger and SR use different versions of JAX-RS that are incompatible. The same approach with using separate classloader is used by the other plugins (kafka, hive and etc.)